### DISCUSSION TITLE
Packet counts not incrementing when we use external encryption engine like DPAA or DPDK

### DISCUSSION THREAD

**Discussion by akky95** [strongswan/strongswan#1802] [Q&A]
Hi,

We are using DPAA in one setup and in another DPDK as an external encryption engine. In both systems we see no increment in packet count. As a result, DPD is being sent even if we receive ESP packets on the system. 

Is there any way or any plugin which handles this case i.e., when we use external encryption engine it get packet count from that ??

Strongswan version : 5.9.1

Thanks, 
Akshay Choudhary

**Comment by tobiasbrunner** [comment]
strongSwan can only use and report statistical data that is actually reported by the IPsec stack and the kernel interface that accesses it (`kernel_ipsec_t::query_sa/policy()`).

**Comment by akky95** [comment]
Thanks Tobias,

Have one more question, looks like DPD is being sent on the basis of last use time of policy get used instead of number of bytes we received...right??

So if we somehow update policy last use time then we it can handle DPD case....right??
